### PR TITLE
[IMP] website_sale: introduce "Collapsible sidebar" and simplify category lists layout

### PR DESCRIPTION
**Pull request by Brieuc-brd**
### Simplify category lists layout 
This PR simplifies the layout of category lists by using the regular link style instead of radio inputs.

### Introduce "Collapsed sidebar" 
This PR introduces the new "Collapsed sidebar" feature. This allows the user to collapse filters and categories to improve the UX when managing a huge amount of filters.

task-3987018

Requires:
- https://github.com/odoo/enterprise/pull/68553

---
### Simplify category lists layout 
| Before | After |
|--------|--------|
| Capture d’écran 2024-08-20 à 09 36 04 | Capture d’écran 2024-08-20 à 09 35 31 |

### Introduce "Collapsed sidebar"
Capture d’écran 2024-09-03 à 08 42 54

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Brieuc-brd** [comment]
Hello @stefanorigano @mano-odoo ! 
The collapsed sidebar is ready for review, could you have a look and tell me what you think ?
Thanks in advance! 🙏

File: addons/website_sale/views/snippets/snippets.xml
Comment (mano-odoo) [inline addons/website_sale/views/snippets/snippets.xml:1]: The collapse category recursive doesn't seem to work anymore 😢 

image
image

File: addons/website_sale/views/templates.xml
Comment (mano-odoo) [inline addons/website_sale/views/templates.xml:842]: This is a nitpick 👼  
```suggestion
            t-attf-class="#{c.id == category.id and 'text-decoration-underline'} p-0"
```

File: addons/website_sale/views/templates.xml
Comment (mano-odoo) [inline addons/website_sale/views/templates.xml:881]: Nitpick with the empty else 👼

**Review by mano-odoo — COMMENTED** [review]
Thanks a lot for handling this tasks 💪 

A few comments below, courage you did a great job 💯

File: addons/website_sale/views/snippets/snippets.xml
Comment (stefanorigano) [reply addons/website_sale/views/snippets/snippets.xml:97]: Debatable... but citing a wise man:

> [...] because users will enable this option only if they have a large amount of items / filters

... and in this scenario chances are that you want always make both collapsible. I'd suggest to follow the specs for now. We can alway adapt the behaviour later on if the use-case proves to be valid 😉

File: addons/website_sale/views/templates.xml
Comment (stefanorigano) [reply addons/website_sale/views/templates.xml:1]: To double-check with @dalaOdoo .
Event if placing categories on top makes more sense almost all the times, I see a couple of business cases where, if renting is active, you want these controllers to be clearly visible and at the very top.

Comment (mano-odoo) [reply addons/website_sale/views/templates.xml:1]: Will be handled in another tasks

Comment (dalaOdoo) [reply addons/website_sale/views/templates.xml:1]: I do believe that if you activate the period filter, it's the most important one.

File: addons/website_sale/views/snippets/snippets.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/snippets/snippets.xml:1]: Good catch 👍 
It's because [this xpath](https://github.com/odoo/odoo/compare/026c44c884d991b97795a74f5f00f70c478705a3..e8add506ad38953f062b59d3fb268db75fcb154a#diff-635040c730ccbb8e4f6f2bf6472cf48a0cc9b8a3d488289dabdfce8096811915R950-R952) targets a `t-attf-class`. But depending if the `Collapsed sidebar` option is triggered,  `website_sale.products_categories_list` or `products_categories_list_collapsed` will be active. 

`products_categories_list_collapsed` used a `t-attf-class` and 
`products_categories_list` used a `class` with `wsale_products_categories_list`, which created the issue.

The only solution I found is to define a `t-attf-class` on `website_sale.products_categories_list` even if there's no function called.

**Comment by Brieuc-brd** [comment]
Hey @mano-odoo 👋 
Thanks for your review 👍 
I've updated the branch, here's the diff :
- COM: https://github.com/odoo/odoo/compare/026c44c884d991b97795a74f5f00f70c478705a3..e8add506ad38953f062b59d3fb268db75fcb154a
- ENT: https://github.com/odoo/enterprise/compare/ffc5adb6251950ab23be5df6e26d94e3498371a6..a73f701756794d4c904fedf302dc1a5d6cea357f

What do you think ?

File: addons/website_sale/views/templates.xml
Comment (mano-odoo) [inline addons/website_sale/views/templates.xml:1087]: Let's wrap the `type` and `multiple` attributes as well

File: addons/website_sale/views/snippets/snippets.xml
Comment (mano-odoo) [reply addons/website_sale/views/snippets/snippets.xml:1]: Interesting, I think you could set the class on both template with a variable for the condition (then xpathing this variable) so we don't erase the `t-attf-` by mistake thinking it's not conditional. I let you decide.

I'm surprised though that even when changing the `class` attribute on your targeted element to a `t-attf-class` it still works with
            `<attribute name="class" add="o_shop_collapse_category" separator=" "/>`
instead of 
`<attribute name="t-attf-class" add="o_shop_collapse_category" separator=" "/>`
🤯

**Review by mano-odoo — COMMENTED** [review]
Thanks again, just one comment and a nitpick😄

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml]: Fixed ✅ 
- Diff 1 : https://github.com/odoo/odoo/compare/18cfe9614d58eeafe7bd099e87b48296de4b878f..feb387f2b6659e94477a42284f8599cd3c03920d#diff-635040c730ccbb8e4f6f2bf6472cf48a0cc9b8a3d488289dabdfce8096811915R1067
- Diff 2 : https://github.com/odoo/odoo/compare/8b379b40e62f4e46ceb31dd1bdd7af1e8a734aca..849ed1d9556fb81ffabcf407c04f9662955afdf3

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml]: Fixed ✅ https://github.com/odoo/odoo/compare/18cfe9614d58eeafe7bd099e87b48296de4b878f..feb387f2b6659e94477a42284f8599cd3c03920d#diff-635040c730ccbb8e4f6f2bf6472cf48a0cc9b8a3d488289dabdfce8096811915R1087-R1089

**Comment by Brieuc-brd** [comment]
@mano-odoo 👋 

> Interesting, I think you could set the class on both template with a variable for the condition (then xpathing this variable) so we don't erase the t-attf- by mistake thinking it's not conditional. I let you decide.

I'm not sure to understand your solution and its benefits 🤔 

> I'm surprised though that even when changing the class attribute on your targeted element to a t-attf-class it still works with
`<attribute name="class" add="o_shop_collapse_category" separator=" "/>`
instead of
`<attribute name="t-attf-class" add="o_shop_collapse_category" separator=" "/>`
🤯

I've made [a change](https://github.com/odoo/odoo/compare/18cfe9614d58eeafe7bd099e87b48296de4b878f..feb387f2b6659e94477a42284f8599cd3c03920d#diff-635040c730ccbb8e4f6f2bf6472cf48a0cc9b8a3d488289dabdfce8096811915L951-R951) to be sure it will always work.

I've also forced the chevron to be always visible for collapsed categories, what do you think ?

Capture d’écran 2024-08-21 à 13 55 11

**Review by mano-odoo — APPROVED** [review]
LGTM

**Comment by dalaOdoo** [comment]
The option to activate it should be after categories and attributes as it depends on these values

**Comment by Brieuc-brd** [comment]
Hey @dalaOdoo 👋 
FYI I moved the option below categories and attributes 👍

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [inline addons/website_sale/views/templates.xml]: Shouldn't you pass `is_sidebar_collapsible` from `website_sale.products` directly?

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [inline addons/website_sale/views/templates.xml]: `is_collapsible`? I know that the option-enabled views have names like something_collapsed, but I think these views, and this current viable, should use "_collapsible".

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [reply addons/website_sale/views/templates.xml:912]: That being said, I fear we'll have to deal with this forced complexity in so many places that it might not be worth spending time making this single template clearer :pensive:

**Review by AntoineVDV — COMMENTED** [review]
Hello :wave:

I just have a few small comments to address, but apart from that it LGTM :)

I pushed a fixup commit with nitpicking changes that you may want to split across your two separate commits (or merge them if you find it simpler). I also did the same on the enterprise branch.

File: addons/website_sale/views/snippets/snippets.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/snippets/snippets.xml]: Indeed, this option is not only for attributes, but also for categories and the web_sale_renting date picker :).

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml:856]: This was made on purpose to avoid a specific issue described here : https://github.com/odoo/odoo/pull/177056#discussion_r1723283881

Perhaps you would have another solution to fix that ? 🙂

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml]: This class was already defined but doesn't seem necessary, I removed it.

**Comment by Brieuc-brd** [comment]
Thanks for your review @AntoineVDV 👍 
I updated the branches and left some comments :)
The diffs :
- COM: https://github.com/odoo/odoo/compare/9a8409496e1366e593746344a7a098d6b3793b15..7797e869357c9b55239829e1d3bb6afd3ba944ed
- ENT: https://github.com/odoo/enterprise/compare/4489e85ef2c96c16de305b88911527ad9af632e6..a102e233314e1f4ae7b0800c98c8a9be9cdb164d

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [reply addons/website_sale/views/templates.xml:856]: Give the divs a `name` and xpath it instead of the classes?

File: addons/website_sale/views/templates.xml
Comment (stefanorigano) [inline addons/website_sale/views/templates.xml:880]: I find the arrows a bit too big, no? I propose
```css
--accordion-btn-icon-width: 1rem;
```

| this PR | proposal |
|--------|--------|
| image | image |

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hey @Brieuc-brd , good job here 👍 
The feature works as expected, I wrote some comments/remarks for you in the code. 
Thanks 💪

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml:955]: @AntoineVDV wdyt ? ☝️

Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml:955]: UPDATE : the current xpath seems to not work when we are in a collapsible sidebar ...
Capture d’écran 2024-08-29 à 09 32 00

So I think we have to push @stefanorigano's solution.

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml:926]: Fixed in temporary commit : https://github.com/odoo/odoo/pull/177056/commits/04817592947f80b0476d0b6f5367ae1468d1bb42

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml]: I applied the changes in a temporary commit to ease the review : https://github.com/odoo/odoo/pull/177056/commits/6126b19dddc02f8761f6c66080ac41bb4435d6e7

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml]: Fixed in this temporary commit : https://github.com/odoo/odoo/pull/177056/commits/04817592947f80b0476d0b6f5367ae1468d1bb42

I also did the same when the `website_sale_renting` date picker is the only item active : https://github.com/odoo/enterprise/compare/9f6b49707470eeb09b292051109d8edcfaf31b1b..312d6ad5d6af7b7e7f0aa40898451abc352011b1

**Comment by Brieuc-brd** [comment]
@stefanorigano 👋 
Thanks for your review 👍 
I've added two temporary commits (I'll squash and fix the indentation once everything is approved) : 
- https://github.com/odoo/odoo/pull/177056/commits/6126b19dddc02f8761f6c66080ac41bb4435d6e7 introduces your fix, thank you for that 🙏 
- https://github.com/odoo/odoo/pull/177056/commits/04817592947f80b0476d0b6f5367ae1468d1bb42 applies others changes related to your review.

Could you take a look ?

@AntoineVDV 👋 Can you also check to be sure everything is approved ?

Thanks in advance !

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [inline addons/website_sale/views/templates.xml:857]: I thought this was a bootstrap class; could you change the name to clarify that it's a placeholder?

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [inline addons/website_sale/views/templates.xml:884]: split line

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [inline addons/website_sale/views/templates.xml:904]: split line

File: addons/website_sale/views/templates.xml
Comment (AntoineVDV) [inline addons/website_sale/views/templates.xml:338]: This is getting really hard to read in diff view:
```suggestion
                                    t-att-class="'products_categories'
                                                 + (' accordion accordion-flush' if is_sidebar_collapsible else ' mb-3')
                                                 + (' border-bottom' if is_sidebar_collapsible and is_view_active('website_sale.products_attributes') else '')"
```

**Review by AntoineVDV — APPROVED** [review]
LGTM once again :smile:

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @Brieuc-brd , all good on my side except just one little design detail.
Almost there, courage 💪

File: addons/website_sale/views/templates.xml
Comment (Brieuc-brd) [reply addons/website_sale/views/templates.xml]: This should be fixed now ✅  What do you think @stefanorigano ? 
The diffs:
- https://github.com/odoo/odoo/compare/3e7b7c976005716bbe5bc31355a911a798bce416..d25d6146caa9fad6fbaef7daf084105da7084994
- https://github.com/odoo/odoo/compare/16065e06c44a85f94a21819fb2ddb89eccb73256..cda4163a457ceaab199f9c5057df626cd6d233e7

**Review by stefanorigano — APPROVED** [review]
LGTM @Brieuc-brd , thanks so much for your work 👍 🙂

**Comment by Brieuc-brd** [comment]
Thanks @stefanorigano 👍 
Squashed and rebased @AntoineVDV 👍

**Comment by AntoineVDV** [comment]
@robodoo r+

**Comment by AntoineVDV** [comment]
@robodoo rebase-ff

**Pull request closed without merging**
